tests/integration/buildcmd/build_integ_base.py

```diff
@@ -396,3 +396,145 @@ def _verify_invoke_built_functions(self, template_path, functions, error_message
                 self.assertIn(error_message, process_stderr)
             else:
                 self.assertEqual(0, process_execute.process.returncode)
+
+
+class CDKTestBase(BuildIntegBase):
```

Comment:
Assuming the test CDK app is written in Python, the test will fail if it is run in a Python environment that `aws_cdk` and other cdk dependencies used in the test projects are not installed. (Same for node.js/typescript or any other languages that CDK supports)

One approach is to add the installation scripts in `appveyor.yaml` (and any other relevant appveyor config) to ensure the dependencies are present in the appveyor environment. But we still need to install the dependencies locally when running the test locally.

An alternative approach (for the Python case) is to create a venv and install the dependencies in the venv. And then use `--cdk-app` option in the command list to instruct `cdk synth` to run within the venv.